src/main/java/org/threadly/concurrent/PriorityScheduler.java

```diff
@@ -1083,131 +997,140 @@ public boolean remove(Runnable task) {
+     * which are NOT ready to run, they will still be included in this total.
      * 
-     * @param task Task to add to end of execute queue
+     * @return Total quantity of tasks queued
      */
-    public void addExecute(OneTimeTaskWrapper task) {
-      executeQueue.add(task);
-
-      handleQueueUpdate();
+    public int queueSize() {
+      return executeQueue.size() + scheduleQueue.size();
     }
 
-    /**
-     * Adds a task for delayed execution.  No safety checks are done at this point.  This call 
-     * will safely find the insertion point in the scheduled queue and insert it into that 
-     * queue.
-     * 
-     * @param task Task to insert into the schedule queue
-     */
-    public void addScheduled(TaskWrapper task) {
+    public void drainQueueInto(List<TaskWrapper> removedTasks) {
+      clearQueue(executeQueue, removedTasks);
       synchronized (scheduleQueue.getModificationLock()) {
-        ClockWrapper.stopForcingUpdate();
-        try {
-          int index = ListUtils.getInsertionEndIndex(scheduleQueue, task, true);
-          
-          scheduleQueue.add(index, task);
-        } finally {
-          ClockWrapper.resumeForcingUpdate();
-        }
+        clearQueue(scheduleQueue, removedTasks);
       }
-      
-      handleQueueUpdate();
     }
-
-    /**
-     * Adds a scheduled task to the end of the scheduled queue.  It is expected that this task is 
-     * NOT ready for execution, and will later be moved from invoking 
-     * {@link #reschedule(RecurringTaskWrapper)}.
-     * 
-     * @param task Task to add to end of schedule queue
-     */
-    public void addScheduledLast(RecurringTaskWrapper task) {
-      scheduleQueue.addLast(task);
-      // no need to notify since this task wont be ready to run
+  
+    private static void clearQueue(Collection<? extends TaskWrapper> queue, List<TaskWrapper> resultList) {
+      Iterator<? extends TaskWrapper> it = queue.iterator();
+      while (it.hasNext()) {
+        TaskWrapper tw = it.next();
+        tw.cancel();
+        if (! (tw.task instanceof ShutdownRunnable)) {
+          int index = ListUtils.getInsertionEndIndex(resultList, tw, true);
+          resultList.add(index, tw);
+        }
+      }
+      queue.clear();
     }
-
-    /**
-     * Call to find and reposition a scheduled task.  It is expected that the task provided has 
-     * already been added to the queue (likely from a call to 
-     * {@link #addScheduledLast(RecurringTaskWrapper)}).  This call will use 
-     * {@link RecurringTaskWrapper#getNextDelayInMillis()} to figure out what the new position 
-     * within the queue should be.
-     * 
-     * @param task Task to find in queue and reposition based off next delay
-     */
-    public void reschedule(RecurringTaskWrapper task) {
-      synchronized (scheduleQueue.getModificationLock()) {
-        if (! workerPool.isShutdownStarted()) {
+    
+    public TaskWrapper getNextTask() {
```

Comment:
Give a look at this logic...this is where we pick which task from each "QueueSet"...basically we figure out if we should have an execute task, or a scheduled task, which ever one is _most_ ready.